spec/requests/demo_mode/sessions_spec.rb

```diff
@@ -72,6 +75,36 @@
           expect(response_json['password']).to be_nil
         end
       end
+
+      context 'with option' do
+        before do
+          DemoMode.configure do
+            around_persona_generation do |_, options|
+              if options.present? && options[:example_custom_option].present?
+                DummyUser.create!(name: options[:example_custom_option][:name])
+              end
+            end
+          end
+        end
+
+        let(:options) { { example_custom_option: { name: 'Tester' } } }
+
+        it 'creates a session and returns processing json saving the option on the created session' do
+          post '/ohno/sessions', params: {
+            session: { persona_name: 'the_everyperson' },
+            options: options,
+          }.to_json, headers: request_headers
+
+          last_session = DemoMode::Session.last
+          DemoMode::AccountGenerationJob.perform_now(last_session, **options)
```

Comment:
Can we use [`#perform_enqueued_jobs`](https://api.rubyonrails.org/v5.2.1/classes/ActiveJob/TestHelper.html#method-i-perform_enqueued_jobs) here instead? The current implementation isn't validating that we enqueue a job that will produce the desired outcome. It's more testing the two steps of the process in isolation, rather than integrated. `#perform_enqueued_jobs` exercises the integrated parts of the system and also doesn't concern itself with the more granular implementation details, like the name of the job.

```suggestion
          perform_enqueued_jobs
```

Reply:
TIL